tests/bgp/test_startup_tsa_tsb_service.py

```diff
@@ -1665,3 +1667,86 @@ def config_reload_linecard_if_unhealthy(lc):
+    5. Verify that the DUT transitions back to the normal state after TSB.
+    6. If the DUT does not return to normal, perform a config reload.
+
+    This test is parameterized with different restart intervals to check
+    service consistency under various conditions.
+    """
+
+    frontend_nodes_per_hwsku = get_frontend_nodes_per_hwsku(duthosts, request)
+    masic_linecard = None
+
+    for lc in frontend_nodes_per_hwsku:
+        if not check_tsa_persistence_support(lc):
+            pytest.skip("TSA persistence not supported in the image")
+
+        if lc.is_multi_asic:
+            masic_linecard = lc
+            break
+
+    if not masic_linecard:
+        pytest.skip("No multi-ASIC linecard found in the testbed")
+
+    tsa_tsb_timer = {}
+    tsa_tsb_timer[masic_linecard] = get_startup_tsb_timer(masic_linecard)
+    if not tsa_tsb_timer[masic_linecard]:
+        pytest.skip("startup_tsa_tsb.service is not supported on {}".format(masic_linecard.hostname))
+
+    initial_tsa_check_before_and_after_test(duthosts)
+
+    try:
+        def restart_startup_tsa_tsb_and_verify(lc, interval):
+            logger.info("Restarting {} startup_tsa_tsb service with interval: {}".format(lc.hostname, interval))
+            lc.shell("systemctl restart {}".format(TSA_TSB_SERVICE))
+            time.sleep(interval)
+            lc.shell('systemctl restart {}'.format(TSA_TSB_SERVICE))
+
+            # for race condition/inconsistent state(where the second restart will be applying TSA to recover),
+            # TSC command check fails the pytest, Hence adding 10 second sleep before TSC is executed
+            time.sleep(10)
+
+            pytest_assert(wait_until(60, 5, 0, get_tsa_tsb_service_status, lc, 'running'),
+                          "startup_tsa_tsb service is not running after restart")
+
+            pytest_assert(wait_until(60, 5, 0,
+                          lambda: get_traffic_shift_state(lc, cmd='TSC no-stats') == TS_MAINTENANCE),
+                          "DUT is not in maintenance state when startup_tsa_tsb service is running")
+
+            lc.shell("TSB")
+            pytest_assert(wait_until(60, 5, 0, get_tsa_tsb_service_status, lc, 'inactive'),
+                          "startup_tsa_tsb service did not stop as expected")
+
+            pytest_assert(wait_until(60, 5, 0, lambda: get_traffic_shift_state(lc, cmd='TSC no-stats') == TS_NORMAL),
+                          "DUT did not return to normal state after executing TSB")
+
+        interval = 0
+        while interval < 10:
+            restart_startup_tsa_tsb_and_verify(masic_linecard, interval)
+            time.sleep(10)
+            interval += 1
+
+    finally:
+        initial_tsa_check_before_and_after_test(duthosts)
+
+        def config_reload_linecard_if_unhealthy(lc):
+            if not (get_traffic_shift_state(lc) == TS_NORMAL):
+                logging.info("DUT's current expected state: {}".format(TS_NORMAL))
```

Comment:
are we logging this as an error/failure?

Reply:
that is a good idea to log this as error/fail the test..
Its not so much applicable to this test specifically, as in each iteration we are explicitly executing TSB and ensuring that DUT returns to Normal(line #1733), so we are covered here.. 
Actually, all the test cases in this file have this similar recovery mechanism, which could be enhanced at once?